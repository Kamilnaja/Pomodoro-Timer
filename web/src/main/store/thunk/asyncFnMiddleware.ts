export async function fetchStats(dispatch: any, getState: any) {
  const response = await client.get("/fakeApi/todos");
  dispatch({ type: "todos/todosLoaded", payload: response.todos });
}
