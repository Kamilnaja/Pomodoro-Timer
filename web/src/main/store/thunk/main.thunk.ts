import TodayStatistics from "../../../shared/store/interfaces/todayStatistics.interface";
import {
  getTodayStatistics,
  getTodayStatisticsError,
  getTodayStatisticsSuccess,
  savePomodoro,
  savePomodoroError,
  savePomodoroSuccess,
} from "../actions/actions";
// todo - different versions for prod etc
const API_URL = "http://localhost:8080/pomodoros";

function makePostRequest() {
  return fetch(API_URL, {
    method: "POST",
  });
}

export function makeGetRequest() {
  return fetch(API_URL);
}

function handleErrors(response: any): Promise<any> {
  if (!response.ok) {
    throw Error(response.statusText);
  }
  return response.json();
}

export function savePomodoroThunk() {
  return (dispatch: Function) => {
    dispatch(savePomodoro());

    return makePostRequest()
      .then(handleErrors)
      .then(() => dispatch(savePomodoroSuccess()))
      .catch((error) => dispatch(savePomodoroError(error)));
  };
}

export function getPomodorosThunk() {
  return (dispatch: Function) => {
    dispatch(getTodayStatistics());

    return makeGetRequest()
      .then(handleErrors)
      .then((payload: TodayStatistics) => {
        console.log(payload);
        return dispatch(getTodayStatisticsSuccess(payload));
      })
      .catch((error) => dispatch(getTodayStatisticsError(error)));
  };
}
