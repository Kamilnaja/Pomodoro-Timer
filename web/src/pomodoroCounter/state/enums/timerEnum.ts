export enum CounterState {
  POMODORO_RUNNING = 'POMODORO_RUNNING',
  BREAK_RUNNING = 'BREAK_RUNNING',
  POMODORO_END = 'POMODORO_END',
  BREAK_END = 'BREAK_END',
  POMODORO_PAUSE = 'POMODORO_PAUSE',
  BREAK_PAUSE = 'BREAK_PAUSE',
}
