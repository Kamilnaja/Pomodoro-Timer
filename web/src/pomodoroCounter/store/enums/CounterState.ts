export enum CounterState {
  RUNNING = 'RUNNING',
  END = 'END',
  PAUSE = 'PAUSE',
  CURRENT_ACTION = 'CURRENT_ACTION',
}
