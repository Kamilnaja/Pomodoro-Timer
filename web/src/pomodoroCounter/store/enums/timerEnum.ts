export enum CounterState {
  POMODORO_RUNNING = 'POMODORO_RUNNING',
  BREAK_RUNNING = 'BREAK_RUNNING',
  POMODORO_END = 'POMODORO_END',
  BREAK_END = 'BREAK_END',
  PAUSE = 'PAUSE',
}
