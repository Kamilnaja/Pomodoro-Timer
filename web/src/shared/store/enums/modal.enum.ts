export enum Modal {
  LOGIN,
  REGISTER,
  SETTINGS,
  NULL,
}
