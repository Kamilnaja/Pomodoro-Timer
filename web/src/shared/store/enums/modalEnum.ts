export enum Modal {
  LOGIN,
  REGISTER,
  SETTINGS,
  LOGOUT,
  NULL,
}
