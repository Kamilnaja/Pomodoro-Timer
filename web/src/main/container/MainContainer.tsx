import { resetForm, setUserIsLoggedIn, setUserIsLoggedOut } from 'auth/store/actions/authActions';
import React from 'react';
import { connect } from 'react-redux';
import { ModalType as ModalEnum } from 'shared/modal/modalEnum';
import { AuthState } from '../../auth/store/interfaces/authState';
import { CloseModal, OpenModal } from '../../shared/modal/ModalProps';
import { MainComponent } from '../component/MainComponent';
import { MainContainerProps } from './MainContainerProps';
import { MainContainerState } from './MainContainerState';

class MainContainer extends React.Component<MainContainerProps, MainContainerState> {
  constructor(props: MainContainerProps) {
    super(props);
    this.state = {
      openedModal: ModalEnum.NULL,
    };
    this.props.setUserIsLoggedIn();
  }

  handleOpenModal: OpenModal = (modal: ModalEnum) =>
    this.setState({
      openedModal: modal,
    });

  handleCloseModal: CloseModal = () => {
    this.setState({
      openedModal: ModalEnum.NULL,
    });
    this.props.resetForm();
  };

  render = () => (
    <MainComponent
      handleOpenModal={this.handleOpenModal}
      handleCloseModal={() => this.handleCloseModal()}
      openedModal={this.state.openedModal}
      isLoggedIn={this.props.auth.isLoggedIn}
    />
  );
}

const mapDispatchToProps = {
  resetForm,
  setUserIsLoggedIn,
  setUserIsLoggedOut,
};

const mapStateToProps = (state: { auth: AuthState }) => {
  const authState = state.auth;
  return { auth: authState };
};

export default connect(mapStateToProps, mapDispatchToProps)(MainContainer);
