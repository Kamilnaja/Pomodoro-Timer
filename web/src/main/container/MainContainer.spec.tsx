import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import MainContainer from './MainContainer';
import thunk from 'redux-thunk';
import { CounterState } from '../../pomodoroCounter/store/enums/CounterState';

const mockStore = configureStore([thunk]);

describe('MainContainer', () => {
  let store;
  let component: renderer.ReactTestRenderer;

  beforeEach(() => {
    store = mockStore({
      auth: {
        isLoggedIn: true,
      },
      pomodoroCounter: {
        counterState: CounterState.RUNNING,
      },
      stats: {
        isLoading: false,
      },
      settings: {
        error: '',
      },
    });

    component = renderer.create(
      <Provider store={store}>
        <MainContainer />
      </Provider>,
    );
  });

  it('Should match the snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
