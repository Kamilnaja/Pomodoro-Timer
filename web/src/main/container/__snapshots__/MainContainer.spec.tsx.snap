// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer Should match the snapshot 1`] = `
<div
  className="app"
>
  <div
    className="container"
  >
    <nav
      className="navbar navbar-expand navbar-light"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        Tomatodone
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="mr-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nav-link"
            data-rb-event-key="/stats"
            href="/stats"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Stats
          </a>
          <a
            className="nav-link"
            data-rb-event-key="/settings"
            href="/settings"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Settings
          </a>
          <a
            className="button--logout nav-link"
            data-rb-event-key={null}
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            Logout
          </a>
        </div>
      </div>
    </nav>
    <div
      className="align-items-center d-flex flex-column bg-dark jumbotron"
    >
      <div
        className="d-flex justify-content-around w-100"
      >
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Pomodoro
        </button>
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Short Break
        </button>
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Long Break
        </button>
      </div>
      <p
        className="time text-light d-flex align-items-center"
      >
        aN:aN
      </p>
      <button
        className="btn--pause btn btn-outline-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Pause timer
      </button>
      <div
        className="info mt-4"
      >
        <div
          className="text-white text-center"
        >
          <span>
            Timer in progress
          </span>
          <p
            className="text-white text-center mt-2"
          >
            Today you have made: 
            <span
              className="badge badge-primary"
            />
             pomodoros
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
